strutil/pathiter.go

```diff
@@ -50,10 +50,10 @@ type PathIterator struct {
 // The path is passed through filepath.Clean automatically.
 func NewPathIterator(path string) (*PathIterator, error) {
 	cleanPath := filepath.Clean(path)
-	if path != cleanPath {
+	if cleanPath != path && cleanPath+"/" != path {
```

Comment:
If you rely on the strict clean path behaviour this had before then maybe adding another constructor eg. `NewAlmostCleanPathIterator()` would be useful.

Reply:
Right now I would never use the more strict constructor. In addition, the less strict constructor can be always used in practice. It is the specific semantic of filepath.Clean that enforces lack of trailing slashes. In practice few system calls care about that.